omod/src/main/java/org/openmrs/module/owa/web/controller/OwaManageController.java

```diff
@@ -48,64 +49,69 @@
-			}
-			
-			model.clear();
-		}
-		return "redirect:manage.form";
-	}
-	
-	@ModelAttribute("settingsValid")
-	public boolean settingsValid() {
-		boolean settingsValid = false;
-		String appFolderPath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_FOLDER_PATH);
-		if (null != appFolderPath) {
-			File file = new File(appFolderPath);
-			if (file.isDirectory() && Files.isWritable(file.toPath())) {
-				settingsValid = true;
-			}
-		}
-		return settingsValid;
-	}
-	
-	@ModelAttribute("appBaseUrl")
-	public String getAppBaseUrl() {
-		String owaBasePath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL,
-		    OwaFilter.DEFAULT_BASE_URL);
-		if (OwaFilter.isFullBasePath(owaBasePath)) {
-			return owaBasePath;
-		} else {
-			return "/openmrs" + owaBasePath;
-		}
-	}
-	
-	@ModelAttribute("appStoreUrl")
-	public String getStoreUrl() {
-		return Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_STORE_URL);
-	}
-}
+	
+	@RequestMapping(value = "/deleteApp", method = RequestMethod.GET)
+	public String deleteApp(@RequestParam("appName") String appName, ModelMap model) {
+		if (appName != null && Context.hasPrivilege("Manage OWA")) {
+			appManager.deleteApp(appName);
+			model.clear();
+		}
+		return "redirect:manage.form";
+	}
+	
+	@RequestMapping(value = "/manager", method = RequestMethod.GET)
+	public String loadSettings(HttpServletRequest request, ModelMap model) {
+		if (Context.hasPrivilege("Manage OWA")) {
+			String appFolderPath = appManager.getAppFolderPath();
+			String appStoreUrl = getStoreUrl();
+			
+			if (null == appFolderPath) {
+				String owaAppFolderPath = OpenmrsUtil.getApplicationDataDirectory()
+				        + (OpenmrsUtil.getApplicationDataDirectory().endsWith(File.separator) ? "owa" : File.separator
+				                + "owa");
+				appManager.setAppFolderPath(owaAppFolderPath);
+			}
+			
+			if (null == appStoreUrl) {
+				appManager.setAppStoreUrl("https://modules.openmrs.org");
+			}
+			
+			model.clear();
+		}
+		return "redirect:manage.form";
+	}
+	
+	@ModelAttribute("settingsValid")
+	public boolean settingsValid() {
+		boolean settingsValid = false;
+		String appFolderPath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_FOLDER_PATH);
+		if (null != appFolderPath) {
+			File file = new File(appFolderPath);
+			if (file.isDirectory() && Files.isWritable(file.toPath())) {
+				settingsValid = true;
+			}
+		}
+		return settingsValid;
+	}
+	
+	@ModelAttribute("appBaseUrl")
+	public String getAppBaseUrl() {
+		String owaBasePath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL,
+		    OwaFilter.DEFAULT_BASE_URL);
+		if (OwaFilter.isFullBasePath(owaBasePath)) {
+			return owaBasePath;
+		} else {
+			return "/openmrs" + owaBasePath;
+		}
+	}
+	
+	@ModelAttribute("allowAdmin")
+	public boolean isAllowAdmin() {
+		return ModuleUtil.allowAdmin();
+	}

```

Comment:
This is the only thing added to `OwaManageController` (+ `ModuleUtil` import).
IDK why `git` thinks I've made revolution here.